### ISSUE TITLE
"NASM not found" :)

### ISSUE THREAD

**Issue by Elyweltonn** [openssl/openssl#19933]
Hi guys

I've been trying everything to build but doesn't matter what I do, once I run "perl Configure" with any parameter, it says "NASM not found - make sure it's installed and availabre on %PATH%"
And yes, it is on %PATH%.
Yes, executing nasm it finds nasm.
I installed VS 2018 and 2022, tried a lot of the version of the compilers.
I installed nasm 32 and then 64 bits to try, tried to execute "vcvarsbat.exe" on 32 and 64 bits.
Rebooted the PC 800 times, tried it in another PC, paste the nasm.exe inside the openssl folder, etc

Those are the versions of nasm, perl and openssl I'm using.

I just don't know how to make it work... I need it to configure "DASH" on Thinkstation P620 Lenovos to be able to manage them remotely.

It sometimes generate the configdata.pm and sometimes doesn't.

Thanks in advance!

Tried to put the configdata.pm here, but couldn't... "Comment is too long"

![Screenshot_50](https://user-images.githubusercontent.com/121051442/208487568-0d261232-f484-4f24-aded-3d7dfc2cfc2d.png)

**Comment by slontis** [comment]
It is likely that your path is still wrong..

You could try debugging the perl to see why..
Search for NASM not found in 10-main.conf and add some perl print statements to see what is going on...

**Comment by t8m** [comment]
> Tried to put the configdata.pm here, but couldn't... "Comment is too long"

Try `./configdata.pm -d`

**Comment by dgm3333** [comment]
Did anyone manage to fix this?
I've tried copying NASM to c:\NASM in case the space in Program Files was causing problems,
copied the path to both user and system environmental variables
run it from standard and elevated command prompts in both VS and vanilla prompt
not sure why this isn't a "bug" - since there are a number of issues raised about it
as below my path definitely has the NASM directory (actually both of them)

C:\Users\dgm55\source\repos\openssl>path
PATH=C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x86;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\VCPackages;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\bin\Roslyn;C:\Program Files\Microsoft Visual Studio\2022\Community\Team Tools\Performance Tools;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools;C:\Program Files\Microsoft Visual Studio\2022\Community\\MSBuild\Current\Bin\amd64;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\;C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\bin\HostX86\x86;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\VCPackages;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\bin\Roslyn;C:\Program Files\Microsoft Visual Studio\2022\Community\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft Visual Studio\Shared\Common\VSPerfCollectionTools\vs2019\;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools;C:\Program Files (x86)\Windows Kits\10\bin\10.0.22000.0\\x86;C:\Program Files (x86)\Windows Kits\10\bin\\x86;C:\Program Files\Microsoft Visual Studio\2022\Community\\MSBuild;D:\StrawberryPerl\perl\bin;C:\NASM;C:\Users\dgm55\AppData\Local\Microsoft\WindowsApps;C:\wxWidgets\3.1.4\lib\vc142_x64_dll;C:\OpenCV\build\x64\vc14\bin;C:\Users\dgm55\AppData\Local\GitHubDesktop\bin;C:\Users\dgm55\.dotnet\tools;C:\Users\dgm55\AppData\Local\Android\Sdk;C:\Users\dgm55\AppData\Local\Android\Sdk\platform-tools;C:\Users\dgm55\AppData\Roaming\npm;C:\Users\dgm55\.dotnet\tools;c:\neuron\bin;c:\neuron\mingw\usr\bin;C:\Users\dgm55\.dotnet\tools;C:\Program Files\NASM;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe

C:\Users\dgm55\source\repos\openssl>perl Configure VC-WIN64A
Configuring OpenSSL version 3.2.0-dev for target VC-WIN64A
Using os-specific seed configuration

Failure!  build file wasn't produced.
Please read INSTALL.md and associated NOTES-* files.  You may also have to
look over your available compiler tool chain or change your configuration.

NASM not found - make sure it's installed and available on %PATH%

**Comment by lorecampioni** [comment]
Hi, I had the same problem and after I tried everything (also moving to openssl-3.1.0 which resulted in other even weirder problems) I had success by switching from ActiveState perl v5.36 to Strawberry perl 5.32.

I hope this helps!

**Comment by pongba** [comment]
> Hi, I had the same problem and after I tried everything (also moving to openssl-3.1.0 which resulted in other even weirder problems) I had success by switching from ActiveState perl v5.36 to Strawberry perl 5.32.
> 
> I hope this helps!

This has solved my problem, thanks!

**Comment by dgm3333** [comment]
@t8m I was already running Strawberry Perl as was the original op so lorecampioni's suggestion to do what we were already doing isn't an answer for our issue.

**Comment by delasy** [comment]
For everyone trying to build it on Windows in GitHub Actions with OpenSSL as ExternalProject_Add in CMake.
You have to add this before checkout:

```yml
- if: runner.os == 'Windows'
  uses: ilammy/msvc-dev-cmd@v1
- if: runner.os == 'Windows'
  uses: ilammy/setup-nasm@v1
```

It solved the issue for me: https://github.com/thelang-io/helpers/commit/8536bc0738a18a64c5143b34837beb1d22b73735

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input

**Comment by JasonStephenson** [comment]
I've encountered this problem and in my case it appears to be related to the error stream redirection

```
my $vc_win64a_info = {};
sub vc_win64a_info {
    unless (%$vc_win64a_info) {
        
        my $no_redirect = `nasm -v`;
        print "before no_redirect\n";
        print $no_redirect;
        print "after no_redirect\n";
        
        my $redirect = `nasm -v 2>NUL`;
        print "before redirect\n";
        print $redirect;
        print "after redirect\n";
        
        if (`nasm -v 2>NUL` =~ /NASM version ([0-9]+\.[0-9]+)/ && $1 >= 2.0) {
```

produces output:
```
C:\Users\JStephenson\source\repos\openssl-3.3.1>perl Configure VC-WIN64A
Configuring OpenSSL version 3.3.1 for target VC-WIN64A
Using os-specific seed configuration
before no_redirect
NASM version 2.16.01 compiled on Jun  1 2023
after no_redirect
before redirect
after redirect
```


i.e with redirection the variable is empty. Dropping the `2>NUL` in the nasm call makes it work as expected
